### ISSUE TITLE
error playlist

### ISSUE THREAD

**Issue by sunry2ntkh** [google/ExoPlayer#3209]
I have problem playlist with set seekTo method
the first I have 5 items for playlist
and current play is position play index 3 of 5 items
and then I want to new items that have 3 items after click and  clear old playlist and add new 3 items to playlist
when I click back back to in old items 3 of 5 items can't not duplicate position play
================Code============================
    private fun exoPlayer(isplay: Boolean, mSourceMedia: ArrayList<Uri>) {
        sourcesPlayList.clear()
        for (url in mDataPlayList.indices) {
            val source = buildMediaSource(mDataPlayList[url])
            sourcesPlayList.add(source)
        }
        println("size playlist: ${sourcesPlayList.size} ")
        try {
            var mediaSource = if (sourcesPlayList.size == 1) sourcesPlayList[0] else ConcatenatingMediaSource(*sourcesPlayList.toTypedArray())
            val haveResumePosition = resumeWindow == C.INDEX_UNSET
            println("seekTo: $resumeWindow $resumePosition $haveResumePosition")
            if (!haveResumePosition) {
                exoPlayer.seekTo(resumeWindow, resumePosition)
            }
            exoPlayer.prepare(mediaSource, !haveResumePosition, false)
            isShowPlayButton = false
            exoPlayer?.playWhenReady = isplay
            mainHandler?.postDelayed(this, TIMER_TRACKING)
            ibFullScreen.setOnClickListener {
                startActivityForResult(buildIntent(this, resumePosition, resumeWindow, resources.getString(R.string.my_lesson_title), mSourceMedia), ConstantUtils.REQUEST_FULLSCREEN_VIDEO)
            }
        } catch (er: Exception) {
            println(er.printStackTrace())
        }
    }

**Comment by tonihei** [comment]
I'm not exactly sure if I understand your question correctly.

What I think you are doing is to use a ConcatenatingMediaSource with 5 items. Then, you play this playlist up to item 3. And then, you like to set up a new playlist with 3 items. 
Your method to set up the player looks fine (and I think it's a copy of our demo app code). 
I'm not sure what you mean by "can't duplicate position play". Could you rephrase the question?

**Comment by sunry2ntkh** [comment]
@tonihei Thank you for your answer! now I am already find the solution with that problem, It is error with the exoplayer initailize singleton pattern after that it not create new initialize it
